source/manual/howto-setup-maintenance-mode.html.md

```diff
@@ -0,0 +1,86 @@
+---
+owner_slack: "#govuk-developers"
+title: Setup maintenance mode
+section: Frontend
+layout: manual_layout
+parent: "/manual.html"
+related_repos: [frontend, government-frontend]
+---
+
+## What is maintenance mode?
+
+Maintenance mode is a way of displaying a user-friendly message on a public facing application to inform them that a service is unavailable e.g. due to planned downtime. This is more helpful than routing them to an error page.
+
+The example below uses the [notice component](https://components.publishing.service.gov.uk/component-guide/notice) to display the message.
+
+## Setup maintenance mode
+
+There are the steps involved:
+
+1. Create a maintenance flag
+2. Disable/Enable E2E(end-to-end) tests
+3. Enable/Disable the maintenance mode in [govuk-helm-charts](https://github.com/alphagov/govuk-helm-charts)
+
+## 1. Create a maintenance flag
+
+Note: This is a one-time setup that is done in the application
+
+1. Add the initializer file for e.g. `maintenance.rb` to read the environment variables under `config/initializers/`
+
+    ```erb
+    Rails.application.config.maintenance_mode = ENV["MAINTENANCE_MESSAGE"].present?
+    Rails.application.config.maintenance_message = ENV["MAINTENANCE_MESSAGE"]
+    ```
+
+### In app view template
+
+This is done when only a part of the application requires downtime.
+
+1. Identify the service that would be impacted
+2. Wrap the code to be shown/hidden in a conditional statement (e.g. `unless`), and include the notice component which uses the `Rails.application.config.maintenance_message`, see [this commit](https://github.com/alphagov/frontend/pull/5072/commits/709494c2d3dc367cea7afe98b7898ea1e0ee19ae) as an example.
+
+   [Example frontend PR](https://github.com/alphagov/frontend/pull/5072)
+
+### In routes file
+
+When we need to ensure that _all routes_ serve the maintenance page during downtime, we can enable the maintenance flag in `routes.rb`. This diverts all incoming traffic to a dedicated maintenance controller, regardless of the request method (GET, POST, etc.)
+
+1. Add the below line in `routes.rb` file to redirect all incoming requests to the `MaintenanceController`:
+
+    ```erb
+    match "*" => "maintenance#show", via: %i[get post] if Rails.application.config.maintenance_mode
+    ```
+
+2. Add the notice component in a `maintenance view` page
+
+    [Example email-alert-frontend PR](https://github.com/alphagov/email-alert-frontend/pull/2159)
+
+## 2. Disable/Enable E2E(end-to-end) tests
+
+During planned outages, E2E (end-to-end) tests — if any — should be disabled to prevent CI/CD pipelines from failing or blocking deployments while the application is unavailable like e.g. [govuk-e2e-tests](https://github.com/alphagov/govuk-e2e-tests/pull/348) and re-enabled once the system downtime is completed.
+
+## 3. Enable/Disable the maintenance mode in govuk-helm-charts
+
+Note: This is done right when you are about to make changes and the system becomes unavailable to users.
```

Comment:
To make this section standout more (and have a grey background) you can update it to have blockquote markdown e.g.
`> Note: This is done right when you are about to make changes and the system becomes unavailable to users.`